src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1094,48 +1094,70 @@
     (dissoc row-map :environment_id)
     row-map))
 
+(defn normalize-resource-event
+  "Prep `event` for comparison/computation of a hash"
+  [event]
+  (-> event
+      (utils/update-when [:timestamp] to-timestamp)
+      (utils/update-when [:old-value] sutils/db-serialize)
+      (utils/update-when [:new-value] sutils/db-serialize)
+      (assoc :containing-class (find-containing-class (:containment-path event)))))
+
+(defn normalize-report
+  "Prep the report for comparison/computation of a hash"
+  [report]
+  (-> report
+      (update-in [:start-time] to-timestamp)
+      (update-in [:end-time] to-timestamp)
+      (update-in [:resource-events] #(map normalize-resource-event %))))
+
+(defn convert-containment-path
+  "Convert the contain path from a collection to the jdbc array type"
+  [event]
+  (utils/update-when event
+                     [:containment-path]
+                     (fn [cp]
+                       (when cp
```

Comment:
n/m, I see now that update-when will call the function if the path ends in `false` or `nil`.